specs/GetEnvironment.md

```diff
@@ -0,0 +1,106 @@
+
+# Description
+Get the `CoreWebView2Environment` used to create the `CoreWebView2` from that `CoreWebView2`'s `Environment` property
+
+
+# Examples
+
+The following code snippet demonstrates how the environment APIs can be use:
+
+## Win32 C++
+
+```cpp
+// Turn on or off image blocking by adding or removing a WebResourceRequested handler
+// which selectively intercepts requests for images.
+void SettingsComponent::SetBlockImages(bool blockImages)
+{
+    if (blockImages != m_blockImages)
+    {
+        m_blockImages = blockImages;
+        if (m_blockImages)
+        {
+            m_webView->AddWebResourceRequestedFilter(L"*", COREWEBVIEW2_WEB_RESOURCE_CONTEXT_IMAGE);
+            CHECK_FAILURE(m_webView->add_WebResourceRequested(
+                Callback<ICoreWebView2WebResourceRequestedEventHandler>(
+                    [this](
+                        ICoreWebView2* sender,
+                        ICoreWebView2WebResourceRequestedEventArgs* args) {
+                        COREWEBVIEW2_WEB_RESOURCE_CONTEXT resourceContext;
+                        CHECK_FAILURE(
+                            args->get_ResourceContext(&resourceContext));
+                        // Ensure that the type is image
+                        if (resourceContext != COREWEBVIEW2_WEB_RESOURCE_CONTEXT_IMAGE)
+                        {
+                            return E_INVALIDARG;
+                        }
+                        // Override the response with an empty one to block the image.
+                        // If put_Response is not called, the request will continue as normal.
+                        wil::com_ptr<ICoreWebView2WebResourceResponse> response;
+                        // Environment Usage
+                        wil::com_ptr<ICoreWebView2Environment> environment;
+                        CHECK_FAILURE(m_webView->get_Environment(&environment));
+                        CHECK_FAILURE(environment->CreateWebResourceResponse(
+                            nullptr, 403 /*NoContent*/, L"Blocked", L"", &response));
+                        CHECK_FAILURE(args->put_Response(response.get()));
+                        return S_OK;
+                    })
+                    .Get(),
+                &m_webResourceRequestedTokenForImageBlocking));
+        }
+        else
+        {
+            CHECK_FAILURE(m_webView->remove_WebResourceRequested(
+                m_webResourceRequestedTokenForImageBlocking));
+        }
+    }
+}
+```
+
+## .NET and WinRT
+
+```c#
+    private void CoreWebView2_WebResourceRequested(object sender, CoreWebView2WebResourceRequestedEventArgs e)
```

Comment:
From another API, but shouldn't the sender be typed?

Reply:
Yes it should.

Reply:
This sample code was written against our .NET API (rather than the WinRT projection into C#) which uses System.EventHandler which seems to generate this style of event handler.

Reply:
Yes. Updated